safe_transaction_service/history/services/transaction_service.py

```diff
@@ -366,25 +373,36 @@ def get_all_txs_from_hashes(
             transfer["token"] = tokens.get(transfer["token_address"])
 
         # Build the list
-        def get_the_transaction(
+        def get_the_transactions(
             transaction_id: str,
-        ) -> Optional[Union[MultisigTransaction, ModuleTransaction, EthereumTx]]:
+        ) -> List[Union[MultisigTransaction, ModuleTransaction, EthereumTx]]:
+            """
+            :param transaction_id:
+            :return: Transactions for the transaction id
+            """
             if result := cached_txs.get(transaction_id):
                 return result
 
             result: Optional[Union[MultisigTransaction, ModuleTransaction, EthereumTx]]
-            if result := multisig_txs.get(transaction_id):
-                # Populate transfers
-                result.transfers = transfer_dict[result.ethereum_tx_id]
+            if result := id_with_multisig_txs.get(transaction_id):
+                for multisig_tx in result:
+                    # Populate transfers
+                    multisig_tx.transfers = transfer_dict[multisig_tx.ethereum_tx_id]
                 return result
 
-            if result := module_txs.get(transaction_id):
-                result.transfers = transfer_dict[result.internal_tx.ethereum_tx_id]
+            if result := id_with_module_txs.get(transaction_id):
+                for module_tx in result:
+                    # Populate transfers
+                    module_tx.transfers = transfer_dict[
+                        module_tx.internal_tx.ethereum_tx_id
+                    ]
                 return result
 
-            if result := plain_ethereum_txs.get(transaction_id):
+            if result := id_with_plain_ethereum_txs.get(transaction_id):
                 # If no Multisig or Module tx found, fallback to simple tx
-                result.transfers = transfer_dict[result.tx_hash]
+                for ethereum_tx in result:
+                    # Populate transfers
+                    ethereum_tx.transfers = transfer_dict[ethereum_tx.tx_hash]
```

Comment:
I have one (maybe silly) doubt here: why are for loops needed for non-module transactions? According to the PR description it seems only module transactions identifiers could have several transactions associated.

Reply:
They are the only ones, but I refactored the dictionaries to all look the same and be `id -> List[Tx]`. I thought that would be more clear than having mixed types and custom logics, and would enable us to allow for example multiple transfers in the same hash or some future features

Reply:
Got it, thanks for the explanation @Uxio0 🙂

Reply:
Thank you for the detailed review 😉